### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by GertjanBisschop**
Update docs on `additional_nodes` flags and correct type in `api.md`.

### CODE REVIEW

File: docs/ancestry.md [hunk 1]
```diff
@@ -958,17 +958,23 @@ do not result in marginal coalescences. For some purposes, however,
...
+{ref}`migration <sec_ancestry_record_migrations>`, 
+and {ref}`pass through <sec_additional_nodes_re>` events. The example below 
+and the next few paragraphs provide a guide on how 
 to record additional nodes and interpret the resulting node tables.
 
 We first set up a simple pedigree simulation. Pedigrees are an
 interesting starting point as in contrast to the coalescent models a single 
-node might be associated with more than one type of event.
+node might be associated with more than one type of event. Conversely, 
+for coalescent models each event is registered as a new node.
```
Comment (petrelharp) [inline docs/ancestry.md:977]:
```suggestion
for the continuous-time coalescent model simulated by default by msprime,
each event is registered as a new node.
```
Couldn't a node in the DTWF be associated with more than one kind of event?

File: docs/ancestry.md [hunk 2]
```diff
@@ -1070,24 +1079,30 @@ nodes across the genealogical history of the sample. By means of bitwise `AND`
...
+
+Note that the `coalescing_segments_only` option should be set to `False` when 
+recording additional nodes. Setting `coalescing_segments_only` to `False` 
+allows us to switch off the default simulator behaviour of only recording the 
+relationships between overlapping ancestry segments. Instead, you can now 
+record edges along the full extent of any of the ancestral lineages 
+that was involved in any of the events as specified by the `additional_nodes` flag. 
+This option can also be set to `False` without specifying any additional nodes. 
+We then register edges along the full length of the tracked ancestral material. 
+Not just for the those marginal overlapping segments. This option will result
```
Comment (petrelharp) [inline docs/ancestry.md:1101]:
```suggestion
We then record edges along the full length of the tracked ancestral material,
not just for the those overlapping segments. This option will result 
```

File: docs/ancestry.md [hunk 3]
```diff
@@ -1096,15 +1111,18 @@ We distinguish two types of common ancestor events: coalescence and common
 ancestor (in the strict sense) events.
 
 Coalescence events are the default node type (associated with value 0) and are 
-therefore only implicitly part of the {class}`.NodeType` enumeration. Whenever we 
-do not observe any marginal overlap between lineages involved in a common ancestor 
-event, we register this in the node table as {class}`msprime.NodeType.COMMON_ANCESTOR`.
+therefore only implicitly part of the {class}`.NodeType` enumeration. 
+In contrast, whenever we do not observe any marginal overlap between the 
+segments of genome carried by the lineages, we register this in the node table as 
+{class}`msprime.NodeType.COMMON_ANCESTOR`.
```
Comment (petrelharp) [inline docs/ancestry.md:1117]:
```suggestion
In contrast, whenever two nonoverlapping ancestral segments
are found to have inherited from the same ancestral genome,
there is no coalescence event, and so we register this in the node table as 
{class}`msprime.NodeType.COMMON_ANCESTOR`.
```

File: docs/ancestry.md [hunk 4]
```diff
@@ -1096,15 +1111,18 @@ We distinguish two types of common ancestor events: coalescence and common
...
 Finally, when keeping track of individuals 
 ({class}``msprime.DiscreteTimeWrightFisher``, 
 {class}``msprime.FixedPedigree``), we might observe genomes (ploid) through
```
Comment (petrelharp) [inline docs/ancestry.md:1120]:
```suggestion
{class}``msprime.FixedPedigree``), we might observe genomes (ploids) through 
```

File: docs/ancestry.md [hunk 5]
```diff
@@ -1096,15 +1111,18 @@ We distinguish two types of common ancestor events: coalescence and common
...
 Finally, when keeping track of individuals 
 ({class}``msprime.DiscreteTimeWrightFisher``, 
 {class}``msprime.FixedPedigree``), we might observe genomes (ploid) through 
-which the ancestral material of only a single lineage passes. This is obviously not a
-common ancestor event. Such genomes can be registered in the table as 
+which the ancestral material of only a single lineage passes. 
+Such genomes can be registered in the table as 
 {class}`msprime.NodeType.PASS_THROUGH` nodes.
+Although, this is obviously not a common ancestor event, remark that for both 
+of these models each node has to carry at least one of these three flags.
```
Comment (petrelharp) [inline docs/ancestry.md:1125]:
```suggestion
Although this is obviously not a common ancestor event, note that for both 
of these models each node has to carry at least one of these three flags.
```

**Comment by petrelharp** [comment]
I made some suggestions to make it more precise; if you see a good way to have an even more precise (and pedantic?) definition of what exactly the node flags record somewhere, that could still be good.

File: docs/ancestry.md (hunk 1 above)
Comment (GertjanBisschop) [reply docs/ancestry.md:977]: yes! good point.

**Comment by GertjanBisschop** [comment]
Thanks a lot @petrelharp!

**Review by jeromekelleher — COMMENTED** [review]
LGTM, happy to merge whenever

**Comment by petrelharp** [comment]
Thanks! Looks good.

**Pull request merged**